packages/web-components/fast-foundation/src/checkbox/checkbox.template.ts

```diff
@@ -18,9 +19,14 @@ export function checkboxTemplate<T extends FASTCheckbox>(
             tabindex="${x => (x.disabled ? null : 0)}"
             @keypress="${(x, c) => x.keypressHandler(c.event as KeyboardEvent)}"
             @click="${(x, c) => x.clickHandler(c.event as MouseEvent)}"
-            class="${x => (x.readOnly ? "readonly" : "")} ${x =>
-                x.checked ? "checked" : ""} ${x =>
-                x.indeterminate ? "indeterminate" : ""}"
+            class="${x =>
+                [
+                    x.readOnly && "readonly",
+                    x.checked && "checked",
+                    x.indeterminate && "indeterminate",
+                ]
+                    .filter(Boolean)
+                    .join(" ")}"
```

Comment:
This makes me think we should have a feature where a class binding can return an object and every key that is truthy gets added a class. Thoughts? Built-in or custom directive?

Reply:
Actually, what if we did something like this:

`class.readonly=${x => x.readOnly}`

We could do this with styles as well:

`style.borderRadius="${x => x.someProperty}px"`

Thoughts?

Reply:
I feel like this moves further away from the "it's just HTML in a tagged template" model that we have right now. `class.readonly` doesn't map directly to any API model - the built-in expectation would probably be closer to `classList` (or `className` which is used in JSX/React) but I feel like that would complicate inference - `class` already exists.

It could also create conflicts, like this:
```
<template class="readonly" class.readonly="${x => x.readOnly}">
```

Reply:
I like the first idea where the `classList` can be managed in the element class rather than in the template. The pro would be that way there's always a state management system, but the con would be that developers would have to explicitly manage that state.

Maybe there's a way to extend the `@attr` decorator to _also_ allow for handling state for a companion class?

Reply:
Mainly I saw the array pattern you were using in the class attribute and wondered if we could do that more efficiently somehow.

Reply:
I think everything but indeterminate can be removed here. Classes are an antipattern on the host and we should delete every instance except this one - which I'll hopefully be sharing a plan to move away from regardless.